k4FWCore/python/k4FWCore/utils.py

```diff
@@ -83,3 +85,43 @@ def load_file(opt_file: Union[TextIOWrapper, str, os.PathLike]) -> None:
     check_wrong_imports(str(code))
 
     exec(code, globals())
+
+
+class LoggingHandler(logging.StreamHandler):
+    def handleError(self, record):
+        # Re-raise SIGPIPE generated exception
+        # from Python 3.11 can be replaced with sys.exception()
+        if isinstance(excp := sys.exc_info()[1], BrokenPipeError):
+            raise excp
+
+        super().handleError(record)
+
+
+_logger = None
+logging.VERBOSE = 5
+logging.addLevelName("VERBOSE", logging.VERBOSE)
+
+LOG_LEVELS = ("VERBOSE", "DEBUG", "INFO", "WARNING", "ERROR")
+
+
+def set_log_level(level: str):
+    """Set the passed log level to the k4run logger"""
+    if level not in LOG_LEVELS:
+        raise ValueError(f"{level} is not a valid log level name")
+    logger = get_logger()
+    logger.setLevel(getattr(logging, level))
+
+
+def get_logger() -> logging.Logger:
+    """Get the logger also used by k4run"""
+    global _logger
+    if _logger is not None:
+        return _logger
+    _logger = logging.getLogger()
+    _logger.setLevel(logging.INFO)
+    formatter = logging.Formatter("[k4run - %(levelname)s] %(module)s.%(funcName)s: %(message)s")
+    handler = logging.StreamHandler(sys.stdout)
```

Comment:
```suggestion
    handler = LoggingHandler(sys.stdout)
```
I'm realizing now that `LoggingHandler` is not used anywhere and indeed piping doesn't work well in all cases:
```
k4run --dry-run ExampleFunctionalProducer.py | head -n 1
```
gives me a bunch of errors like:
```
[k4run] --> ExampleFunctionalProducer
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.13/logging/__init__.py", line 1154, in emit
    self.flush()
    ~~~~~~~~~~^^
  File "/usr/lib/python3.13/logging/__init__.py", line 1136, in flush
    self.stream.flush()
    ~~~~~~~~~~~~~~~~~^^
BrokenPipeError: [Errno 32] Broken pipe
Call stack:
  File "/k4FWCore/install/bin/k4run", line 258, in <module>
    main()
  File "/k4FWCore/install/bin/k4run", line 213, in main
    logger.info(f"Option name: {optionName} {opts_dict[optionName]}")
Message: 'Option name: IOSvc.CollectionNames []'
```

Reply:
Fixed in https://github.com/key4hep/k4FWCore/pull/313